packages/frontend/src/components/common/modal/ChartCreateModal/SaveToSpaceOrDashboard.tsx

```diff
@@ -0,0 +1,428 @@
+            id="select-dashboard"
+            label="Dashboard"
+            size="xs"
+            data={dashboards.map((d) => ({
+                value: d.uuid,
+                label: d.name,
+                group: (spaces ?? []).find((s) => s.uuid === d.spaceUuid)?.name,
+            }))}
+            rightSection={isLoadingDashboards && <Loader size="xs" />}
+            defaultValue={
+                dashboards.find((d) => d.spaceUuid === form.values.spaceUuid)
+                    ?.uuid
+            }
+            searchable
+            nothingFound="No matching dashboards found"
+            filter={(value, dashboard) =>
+                !!dashboard.label
+                    ?.toLowerCase()
+                    .includes(value.toLowerCase().trim())
+            }
+            withinPortal
+            required={form.values.saveDestination === SaveDestination.Dashboard}
+            {...form.getInputProps('dashboardUuid')}
+        />
+    );
+};
+
+type SaveToSpaceOrDashboardProps = {
+    savedData: CreateSavedChartVersion;
+    defaultSpaceUuid: string | undefined;
+    dashboardInfoFromSavedData: {
+        dashboardUuid: string | null;
+        dashboardName: string | null;
+    };
+    onConfirm: (savedData: CreateSavedChartVersion) => void;
+    onClose: () => void;
+};
+
+export const SaveToSpaceOrDashboard: FC<SaveToSpaceOrDashboardProps> = ({
+    savedData,
+    defaultSpaceUuid,
+    dashboardInfoFromSavedData,
+    onConfirm,
+    onClose,
+}) => {
+    const { projectUuid } = useParams<{ projectUuid: string }>();
+
+    const { mutateAsync: createChart } = useCreateMutation();
+
+    const [, setShouldCreateNewSpace] = useState(false);
+    const { mutateAsync: createSpace } = useSpaceCreateMutation(projectUuid);
+
+    const form = useForm<ChartCreateModalFormValues>({
+        initialValues: {
+            name: '',
+            spaceUuid: '',
+            dashboardUuid: '',
+            dashboardName: '',
+            description: '',
+            newSpaceName: null,
+            saveDestination: SaveDestination.Space,
+        },
+
+        validate: {
+            name: (value) => (value.length > 0 ? null : 'Name is required'),
+        },
+    });
+
+    const { data: dashboards, isInitialLoading: isLoadingDashboards } =
+        useDashboards(
+            projectUuid,
+            {
+                staleTime: 0,
+                onSuccess: () => {
+                    if (
+                        dashboardInfoFromSavedData.dashboardUuid &&
+                        dashboardInfoFromSavedData.dashboardName
+                    ) {
+                        form.setFieldValue(
+                            'dashboardUuid',
+                            dashboardInfoFromSavedData.dashboardUuid,
+                        );
+                        form.setFieldValue(
+                            'dashboardName',
+                            dashboardInfoFromSavedData.dashboardName,
+                        );
```

Comment:
you can call `form.initialize(` and pass down initial field values.

you can distinguish between initialized/un-initialized forms using `form.initialized`

Reply:
done!